File: typescript/Behavioral/16-Mediator/Colleague/BillingService.ts
import { Colleague } from "./Colleague";
import { IDispatchMediator } from "../Mediator/IDispatchMediator";

export class BillingService extends Colleague {
  constructor(mediator: IDispatchMediator) {
    super(mediator, "BillingService");
  }

  generateInvoice(jobId: string, handyman: string): void {
    console.log(
      `  [BillingService] Invoice generated for Job #${jobId} completed by ${handyman}.`,
    );
  }
}

File: csharp/Behavioral/16-Mediator/Colleague/BillingService.cs
public class BillingService : Colleague
{
    public BillingService(IDispatchMediator mediator)
        : base(mediator, "BillingService") { }

    public void GenerateInvoice(string jobId, string handyman)
    {
        Console.WriteLine($"  [BillingService] Invoice generated for Job #{jobId} " +
                          $"completed by {handyman}.");
    }
}
